pkg/chunked/dump/dump.go

```diff
@@ -198,10 +199,14 @@ func dumpNode(out io.Writer, added map[string]*internal.FileMetadata, links map[
 		return err
 	}
 
-	for k, v := range entry.Xattrs {
+	for k, vEncoded := range entry.Xattrs {
+		v, err := base64.StdEncoding.DecodeString(vEncoded)
+		if err != nil {
+			return fmt.Errorf("decode xattr %q: %w", k, err)
+		}
 		name := escaped(k, ESCAPE_EQUAL)
-		value := escaped(v, ESCAPE_EQUAL)
 
+		value := escaped(string(v), ESCAPE_EQUAL)
```

Comment:
I think this error would have been more obvious if `escaped()` took a `[]byte` right? That's what we have in[the rust code](https://github.com/containers/composefs/blob/45e6179993aa8a057d27946e26f5f7306b6a1779/rust/composefs/src/dumpfile.rs#L173). [The C code](https://github.com/containers/composefs/blob/45e6179993aa8a057d27946e26f5f7306b6a1779/tools/composefs-info.c#L56)...well, a bit like Go, "byte array" vs "string" is a fluid, loosely enforced concept, but the presence there of a "length" member does signal it's binary.